### PR TITLE
doc: add hld-security guest secure boot description

### PR DESCRIPTION
**Pull request by yonghuah**
1. add guest secure boot with OVMF.
  2. delete obsolete content.
  3. other minor update.

Signed-off-by: Yonghua Huang <yonghua.huang@intel.com>

### CODE REVIEW

**Comment by yonghuah** [comment]
@fyin1  @peterfang  can you please review this PR? thanks.

**Comment by acrnsi** [comment]
OK to verify

**Comment by yonghuah** [comment]
@peterfang  have updated this PR, thanks for comments.

**Comment by yonghuah** [comment]
@fyin1  can review this PR? thanks
-Yonghua

**Comment by NanlinXie** [comment]
@yonghuah, Please update SOS -> Service VM and UOS -> User VM.

**Comment by peterfang** [comment]
@yonghuah Looks good to me. Thanks!

**Comment by acrnsi** [comment]
OK to verify

File: doc/developer-guides/hld/hld-security.rst [hunk 1]
```diff
@@ -193,6 +193,50 @@ partners are responsible for image signing, ensuring the key strength
...
+the UEFI Secure Boot Policy, the platform manufacturer or the platform owner enroll the
+policy objects, which include the n-tuple of keys {PK, KEK, db,dbx} as a step 1.
+During each successive boot, the UEFI secure boot implementation will assess the
+policy in order to verify the signed images that are discovered in a host-bus adapter
+or on a disk. If the images pass policy, then they are invoked.
+
+In ACRN, User VM Secure Boot can be enabled by below steps.
+
+#. Generate keys(PK/KEK/DB) with key generation tool such as Ubuntu KeyGeneration,
+   `PK.der`, `KEK.der` and `db.der` will be enrolled in UEFI BIOS, `db.key` and `db.crt`
```
Comment (xyzhao2018) [inline doc/developer-guides/hld/hld-security.rst]: Previously, db.der is used. A little confused about db.key and db.crt, if it is the different format of db, please introduce it before.

**Comment by acrnsi** [comment]
OK to verify

**Comment by acrnsi** [comment]
OK to verify

**Comment by acrnsi** [comment]
OK to verify

**Comment by yonghuah** [comment]
Please hold on, update is required for this PR.

**Comment by yonghuah** [comment]
@NanlinXie  Update done, please review, thanks.

**Comment by yonghuah** [comment]
@xyzhao2018  sorry for late update, can you review? thanks.

File: doc/developer-guides/hld/hld-security.rst (hunk 1 above)
Comment (xyzhao2018) [reply doc/developer-guides/hld/hld-security.rst]: Looks good to me, Thanks!

**Pull request merged**